### DISCUSSION TITLE
Welcome to bytemask Discussions!

### DISCUSSION THREAD

**Discussion by PatilShreyas** [PatilShreyas/bytemask#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ernesto-olalla** [comment]
Can I use it in a Kotlin multiplatfrom project? I need to use it in shared module

**Reply by PatilShreyas** [reply]
Not as of now

**Comment by M1sha0** [comment]
Hi! I’m application security engineer, and I have a question. Isn’t more secure to have only signing info to encrypt data, without having choice between signing info and custom key? I think in this way because if someone will try to decompile your apk, he just need to find config and change there encryption on signing info to custom key and set custom key value to our singing info ( because it’s public info)

**Reply by PatilShreyas** [reply]
Correct, but sometime developers might have a use case of providing signing key via some external sources (like environment variables in CI, some non-VCS committed files, etc) then this can be a blocker. So, this is exposed in order to support this and let developers choose how they want to provide the signing info.

**Reply by M1sha0** [reply]
Can you give an example of providing signing key via external source?

**Reply by PatilShreyas** [reply]
Example: keeping signing info in the environment variables. So feeding SHA key via environment variable

**Reply by M1sha0** [reply]
> keeping signing info in the environment variables. So feeding SHA key via environment variable

but its strange logic. You always have acess to signature of APK, and there is not need of custom encryption key. I like your plugin, and want to improve security and popularity of it, and i think better to delete permission to have custom key.

**Comment by M1sha0** [comment]
Also I don’t understand why you add properties file in apk, and give to attacker functionality to change the way of decryption. I think better just delete properties file, and always have system call to signature, without having ability to change the key.

**Reply by PatilShreyas** [reply]
It's just part of source code, not APK

**Reply by M1sha0** [reply]
> It's just part of source code, not APK

Oh, maybe i did something wrong... So, as i understand there is no config file in apk, then how application decrypt code while runtime?
